samples/smpe_list.py

```diff
@@ -0,0 +1,181 @@
+
+    try:
+        # Setup base DDs
+        dd_list.append(DDStatement("SMPCSI",
+                                   DatasetDefinition(
+                                       defaults["SMPECSI"]["Dataset"])))
+        dd_list.append(DDStatement("SMPLOG", "DUMMY"))
+        dd_list.append(DDStatement("SMPLOGA", "DUMMY"))
+
+        # Create Temporary File
+        temp_dataset_name = datasets.tmp_name(high_level_qualifier)
+        temp_dataset = datasets.create(temp_dataset_name, type="PDS",
+                                       primary_space=(defaults["TEMP_DATASET"]["Primary_space"]).strip(),
+                                       secondary_space=(defaults["TEMP_DATASET"]["Secondary_space"]).strip(),
+                                       block_size=3200,
+                                       record_format="FB", record_length=80,
+                                       volumes=(defaults["TEMP_DATASET"]["Volume"]).strip(),
+                                       directory_blocks=10)
+
+        # add it to the ddList
+        dd_list.append(DDStatement("SMPWRK6",
+                                   DatasetDefinition(temp_dataset_name)))
+
+        # define the input for the program make - sure the input is EBCIDIC
+        sysin_file_name = defaults["SMPECNTL"]["Filename"]
+        with open(sysin_file_name, mode="w", encoding="cp1047") as file:
+            file.write(f"SET     BDY({target_zone}).\n")
+            if list_options is None:
+                file.write("LIST.\n")
+            else:
+                file.write(f"LIST {list_options}.\n")
+        dd_list.append(DDStatement("SMPCNTL",
+                                   FileDefinition(sysin_file_name)))
+
+        # define the place for the output to go
+        output_dataset_name = datasets.tmp_name(high_level_qualifier)
+        _ = datasets.create(output_dataset_name, type="SEQ",
+                            primary_space=(defaults["OUTPUT_DATASET"]["Primary_space"]).strip(),
+                            secondary_space=(defaults["OUTPUT_DATASET"]["Secondary_space"]).strip(),
+                            volumes=(defaults["OUTPUT_DATASET"]["Volume"]).strip())
+        dd_list.append(DDStatement("SMPLIST",
+                                   DatasetDefinition(output_dataset_name)))
+
+        # execute the program
+        command_return_code = mvscmd.execute_authorized(pgm="GIMSMP",
+                                                        dds=dd_list)
+
+    except Exception as e:
+        sys.stderr.write("Error processing command environment...\n")
+        sys.stderr.write(f"Exception information: {e}\n")
+        sys.exit(1)
+
+    finally:
+        # remove temporary dataset and file
+        if temp_dataset:
+            datasets.delete(temp_dataset_name)
+        os.remove(sysin_file_name)
+
+    print(f"Output can be found in: {output_dataset_name}\n")
+
+    return command_return_code 
+
+
+def getdefaults(filename):
```

Comment:
I would change this from `getdefaults` to `get_defaults` to preserve PEP8 standards.